### DISCUSSION TITLE
escaping placeholder values in filters

### DISCUSSION THREAD

**Discussion by jkdoshi** [pocketbase/pocketbase#1495] [Q&A] [answered]
I'm using the JS-SDK and calling ...

```js
  const records = await client.collection("auditlog").getFullList(20, {
    filter: `collection='${collection}' && record='${id}'`,
  });
```

Where `collection` and `id` are string variables. The problem with this code is it will break if there are any special characters (e.g. single-quote `'`) in either of those two variables. How to make sure such characters are escaped properly?

Other API's handle this by letting the caller supply filter as an  object with params as key-values instead of the entire filter as a string. Like so ...

```js
  const records = await client.collection("auditlog").getFullList(20, {
    filter: {collection:collection, record:id},
  });
```

Maybe I'm missing something obvious.

**Comment by ganigeorgiev** [comment, marked as answer]
> How to make sure such characters are escaped properly?

You have to escape them manually, eg. prefixing the js variable value single quotes with `\`. For now there is no other solution.

A simple key-value object will not work in our case because the filters are a little more complicated - you have other operators in addition to `=`, nested grouping, option to combine expressions with `&&`/`||`, option to compare operands with identical source (eg. columnVScolumn, valueVSvalue), etc.

I have it in my todo to add support for a helper "builder" object that will take care for constructing the final filter expressions, but for now it is a very low priority. Something like:

```js
const records = await pb.collection("auditlog").getFullList(20, {
  filter: pb.filter().eq("collection", collection).eq("record", id).build(),
});
```
(_the syntax most likely will not be this_)

**Comment by egnor** [comment]
In case anyone else lands here: https://github.com/pocketbase/js-sdk#binding-filter-parameters
